internal/internal_command_state_machine.go

```diff
@@ -905,13 +874,21 @@ func (h *commandsHelper) setCurrentWorkflowTaskStartedEventID(workflowTaskStarte
 	// corresponding history event after processing. So we can use workflow task started event id + 2 as the offset as
 	// workflow task completed event is always the first event in the workflow task followed by events generated from
 	// commands. This allows client sdk to deterministically predict history event ids generated by processing of the
-	// command. We must also add the number of cancel commands that were spawned during cancellation of the workflow
-	// execution as those canceled command events will show up *after* the workflow task completed event.
-	h.nextCommandEventID = workflowTaskStartedEventID + 2 + h.commandsCancelledDuringWFCancellation
-	h.commandsCancelledDuringWFCancellation = 0
-	// We must change the counter here so that others who mutate
-	// commandsCancelledDuringWFCancellation know it has since been reset
-	h.nextCommandEventIDResetCounter++
+	// command. It is possible, notably during workflow cancellation, that commands are generated before the workflow
+	// task started event is processed. In this case we need to adjust the nextCommandEventID to account for these unsent
+	// commands.git
+	var uncountedCommands int64
+	for curr := h.orderedCommands.Front(); curr != nil; {
+		d := curr.Value.(commandStateMachine)
+		command := d.getCommand()
```

Comment:
I wish this call were `createCommand()`. It was unclear to me what was happening here until I understood that's what this is.